server/lib/models/workspace.ts

```diff
@@ -208,6 +208,11 @@ export default class Workspace extends Model<Workspace> {
       // not the pointer containing the honest answer candidate.
       // This pointer is contained in the malicious question.
       const maliciousWorkspace = await Workspace.findByPk(workspace.parentId);
+      if (!maliciousWorkspace) {
```

Comment:
I would chance this condition to `!maliciousWorkspace || !maliciousWorkspace.isEligibleForMaliciousOracle)`. So now even if a workspace has a parent, if this parent isn't malicious then we stop here.